File: python_solutions/2962.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        max_val = max(nums)
        max_loc = []
        max_loc.append(-1)
        for idx,num in enumerate(nums):
            if num == max_val:
                max_loc.append(idx)
        max_loc.append(len(nums))
        arr_cnt = 0
        for idx in range(1,len(nums)+1):
            if idx + k-1 < len(max_loc):
                arr_cnt += (max_loc[idx]-max_loc[idx-1]) * (len(nums)-max_loc[idx+k-1])
        return arr_cnt
        

File: rust_solutions/2962.rs
impl Solution {
    pub fn count_subarrays(nums: Vec<i32>, k: i32) -> i64 {
        let nums_size = nums.len();
        let max_val = *nums.iter().max().unwrap() ;
        let mut arr_cnt : i64 = 0 ;
        let mut max_loc:Vec<usize> = Vec::new();
        max_loc.push(0);
        for (idx,value) in nums.iter().enumerate() {
            if *value == max_val {
                max_loc.push(idx+1);
            }
        }
        max_loc.push(nums_size+1);
        let add_idx : usize = k as usize - 1;
        for idx in 1..nums_size+1 {
            match max_loc.get(idx + add_idx) {
                Some(&next_loc) => {
                    arr_cnt += (max_loc[idx]-max_loc[idx-1]) as i64 * (nums_size-next_loc+1) as i64 ;
                },
                None => break
            }
        }
        arr_cnt
        
    }
}
